### PR TITLE
Adding Long factorials

### PR DESCRIPTION
**Pull request by Saviour1001**
I have read the contributor guidelines and have worked as per them. 

### Description

The algorithm of finding long factorials is already present in the algorithm database but isn't that understandable. It uses a lot of inbuilt functions and thus is hard to follow. I have added a different way to find the output and have explained it much better. Thus I have enhanced the existing algorithm. 

### Checklist

- [ YES ] I've read the contribution guidelines.
- [ YES ] I've checked the issue list before deciding what to submit.
- [ YES ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

I have added a file with a new algorithm to solve the question and have made changes to the Readme where I have added the link to the question. 
Github

### CODE REVIEW

**Comment by Saviour1001** [comment]
This PR Closes issue #2452

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -187,8 +188,8 @@ _add list here_
 - [Longest common sub sequence](dp/longest_increasing_subsequence.cpp)
 - [Longest increasing subsequence](dp/longest_increasing_subsequence.cpp)
 - [Matrix Chain Multiplication](dp/Matrix_Chain_Multiplication.cpp)
-- [Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)
-- [[Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)]
+- [Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)
+- [[Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)]
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Revert these changes

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+This program here uses a vector to store each particular digit in a single index of the vector. 
+The carry is forwarded as basic maths and in the end the complete result gets stored in the vector*/
+
+#include <vector>
+#include <iostream>
+
+using namespace std;
+
+
+int main() {
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/long_factorials.cpp]: Use main() only for the driver code. Use some other function for the main code logic

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    int num;
+    int carry = 0;       
+    cin >> num;                  //Input the number whose factorial is to be calculated 
+    vector <int> arr(10000, 0);  //The size of the vector should be more than the digits present 
+    arr[0] = 1;                  //in the factorial of the number.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/long_factorials.cpp]: Use comments above the line of code not beside it. Also don't use redundant comments

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    arr[0] = 1;                  //in the factorial of the number.  
+    int k = 0;                         
+
+    for(int i = 1; i <= num; i++) {
+        for(int j = 0;j <= k; j++) {      
+            arr[j] = arr[j] * i + carry;
+            carry = arr[j] / 10;
+            arr[j] = arr[j] % 10;
+        }
+        while(carry) {        //The carries are forwarded in order to get the final result
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/long_factorials.cpp]: Comments above the line of code

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            k++;
+            arr[k] = carry % 10;
+            carry /= 10;
+        }   
+    }
+    for(int i = k; i >= 0; i--) {  //Each digit of the final answer is stored in a single index 
+        cout << arr[i];            //of the vector
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/long_factorials.cpp]: Comments above the line of code

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+            arr[k] = carry % 10;
+            carry /= 10;
+        }   
+    }
+    for(int i = k; i >= 0; i--) {  //Each digit of the final answer is stored in a single index 
+        cout << arr[i];            //of the vector    
+    }
+    cout << "\n";
+    return 0;
+}
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/long_factorials.cpp:49]: Add sample I/O and complexity of the Code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/long_factorials.cpp (hunk 2 above)
Comment (Saviour1001) [reply C-Plus-Plus/cp/long_factorials.cpp]: Done

File: C-Plus-Plus/cp/long_factorials.cpp (hunk 6 above)
Comment (Saviour1001) [reply C-Plus-Plus/cp/long_factorials.cpp:49]: Mentioned them

File: C-Plus-Plus/cp/long_factorials.cpp (hunk 5 above)
Comment (Saviour1001) [reply C-Plus-Plus/cp/long_factorials.cpp]: Removed the comments beside the code

File: C-Plus-Plus/cp/long_factorials.cpp (hunk 3 above)
Comment (Saviour1001) [reply C-Plus-Plus/cp/long_factorials.cpp]: Removed the redundant comments

**Comment by Saviour1001** [comment]
Managed to solve all the asked changes and updated the PR as per them @HarshCasper

File: C-Plus-Plus/README.md [hunk 7]
```diff
@@ -187,8 +187,8 @@ _add list here_
 - [Longest common sub sequence](dp/longest_increasing_subsequence.cpp)
 - [Longest increasing subsequence](dp/longest_increasing_subsequence.cpp)
 - [Matrix Chain Multiplication](dp/Matrix_Chain_Multiplication.cpp)
-- [Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)
-- [[Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)]
+- [Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)
+- [Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Revert this

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    factorial(num);
+    return 0;
+}
+
+/*Sample test cases 
+1. Input: 5
+   Output: 120
+2. Input: 50
+   Output: 30414093201713378043612608166064768844377641568960512000000000000 
+ */
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/long_factorials.cpp:59]: Add a new line at the EOF

Comment (HarshCasper) [inline C-Plus-Plus/cp/long_factorials.cpp]: Remove the second I/O sample and instead add Time-Space Complexity

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+// Time Complexity = O(N)
+
+#include <vector>
+#include <iostream>
+
+using namespace std;
+
+
+void factorial(int val)
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/long_factorials.cpp:20]: Add function description

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Saviour1001** [comment]
Got the changes that were asked for, but then I don't know what went wrong with the DeepSource Java Test. @HarshCasper

File: C-Plus-Plus/cp/long_factorials.cpp (hunk 8 above)
Comment (HarshCasper) [reply C-Plus-Plus/cp/long_factorials.cpp:59]: @Saviour1001 Please resolve this
